hooks/gen_docs.py

```diff
@@ -1,10 +1,317 @@
 """Generates the whole 'generatable' KPOps documentation"""
+import logging
 import subprocess
+import sys
+from pathlib import Path
+from typing import NamedTuple, cast
+
+import typer
+import yaml
 
 from hooks import PATH_ROOT
+from kpops.cli.registry import _find_classes
+from kpops.components import KafkaConnector, PipelineComponent
+from kpops.utils.yaml_loading import load_yaml_file
 
 PATH_KPOPS_MAIN = PATH_ROOT / "kpops/cli/main.py"
 PATH_CLI_COMMANDS_DOC = PATH_ROOT / "docs/docs/user/references/cli-commands.md"
+PATH_DOCS_RESOURCES = PATH_ROOT / "docs/docs/resources"
+PATH_DOCS_COMPONENTS = PATH_DOCS_RESOURCES / "pipeline-components"
+PATH_DOCS_COMPONENTS_DEPENDENCIES = (
+    PATH_DOCS_COMPONENTS / "dependencies/pipeline_component_dependencies.yaml"
+)
+PATH_DOCS_COMPONENTS_DEPENDENCIES_DEFAULTS = (
+    PATH_DOCS_COMPONENTS / "dependencies/defaults_pipeline_component_dependencies.yaml"
+)
+PATH_DOCS_KPOPS_STRUCTURE = PATH_DOCS_COMPONENTS / "dependencies/kpops_structure.yaml"
+KPOPS_COMPONENTS = tuple(_find_classes("kpops.components", PipelineComponent))
+KPOPS_COMPONENTS_INHERITANCE_REF = {
+    component.get_component_type(): cast(
+        type[PipelineComponent], component.__base__
+    ).get_component_type()
+    for component in KPOPS_COMPONENTS
+}
+KPOPS_COMPONENTS_SECTIONS = {
+    # `set` to make it pickleable
+    component.get_component_type(): set(component.__fields__.keys())
+    for component in KPOPS_COMPONENTS
+}
+KPOPS_COMPONENTS_SECTIONS_ORDER = [
+    "type",
+    "name",
+    "namespace",
+    "app",
+    "from_",
+    "to",
+    "prefix",
+    "repo_config",
+    "version",
+    "resetter_values",
+    "offset_topic",
+]
+
+log = logging.getLogger("DocumentationGenerator")
+
+#####################
+# EXAMPLES          #
+#####################
+
+DANGEROUS_FILES_TO_CHANGE = {
+    PATH_DOCS_COMPONENTS_DEPENDENCIES,
+    PATH_DOCS_COMPONENTS_DEPENDENCIES_DEFAULTS,
+    PATH_DOCS_KPOPS_STRUCTURE,
+}
+# All args provided to the script
+# Pre-commit passes changed files as args
+SYS_ARGV = set(sys.argv)
```

Comment:
```suggestion
CHANGED_FILES = set(sys.argv)
```
?

Reply:
I am not sure if that would be correct. Anything could be passed to the script as an arg and it would end up in that constant.

How about `SCRIPT_ARGUMENTS`?